### PR TITLE
Remove adding package name to the `Ballerina.toml`

### PR DESCRIPTION
**Pull request by pramodya1994**
- Fixes https://github.com/ballerina-platform/ballerina-lang/issues/36151

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java [hunk 1]
```diff
@@ -247,29 +258,47 @@ private PackageDescriptor getPackageDescriptor(TomlTableNode tomlTableNode) {
...
         if (topLevelPkgNode == null || topLevelPkgNode.kind() != TomlType.TABLE) {
+            reportDiagnostic(MISSING_PKG_INFO_ERROR,
+                    ProjectDiagnosticErrorCode.MISSING_PKG_INFO_IN_BALLERINA_TOML.diagnosticId(),
+                    DiagnosticSeverity.WARNING, tomlTableNode.location());
             return PackageDescriptor.from(defaultOrg(), defaultName(this.projectPath), defaultVersion());
         }
 
         TomlTableNode pkgNode = (TomlTableNode) topLevelPkgNode;
         if (pkgNode.entries().isEmpty()) {
+            reportDiagnostic(MISSING_PKG_INFO_ERROR,
```
Comment (sameerajayasoma) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java:277]: I think we need to inform the user of the default values inferred from the context with all these error messages.

**Review by sameerajayasoma — CHANGES_REQUESTED** [review]

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java (hunk 1 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java]: Added via https://github.com/ballerina-platform/ballerina-lang/pull/36174/commits/2e9a0e40f72adcbdce684f1f6b8b6ec354508d4f

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java [hunk 2]
```diff
@@ -257,49 +255,53 @@ private PackageDescriptor getPackageDescriptor(TomlTableNode tomlTableNode) {
...
-            reportDiagnostic(MISSING_PKG_INFO_ERROR,
+            reportDiagnostic(errorMessage,
                     ProjectDiagnosticErrorCode.MISSING_PKG_INFO_IN_BALLERINA_TOML.diagnosticId(),
                     DiagnosticSeverity.WARNING, tomlTableNode.location());
             return PackageDescriptor.from(defaultOrg(), defaultName(this.projectPath), defaultVersion());
         }
 
         TomlTableNode pkgNode = (TomlTableNode) topLevelPkgNode;
         if (pkgNode.entries().isEmpty()) {
-            reportDiagnostic(MISSING_PKG_INFO_ERROR,
```
Comment (sameerajayasoma) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java]: I don't think we need this condition. It is covered by the org, name, and version checks below

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java [hunk 3]
```diff
@@ -257,49 +255,53 @@ private PackageDescriptor getPackageDescriptor(TomlTableNode tomlTableNode) {
...
                     ProjectDiagnosticErrorCode.MISSING_PKG_INFO_IN_BALLERINA_TOML.diagnosticId(),
                     DiagnosticSeverity.WARNING, pkgNode.location());
             return PackageDescriptor.from(defaultOrg(), defaultName(this.projectPath), defaultVersion());
         }
 
         org = getStringValueFromTomlTableNode(pkgNode, "org");
         if (org == null) {
-            reportDiagnostic("missing 'org' under [package]",
+            org = defaultOrg().value();
+            reportDiagnostic("missing 'org' under [package], value derived as '" + org + "'",
```
Comment (sameerajayasoma) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java:288]: Here is my suggestion:

`missing key 'org' in table '[package]' in 'Ballerina.toml'. Defaulting to 'org = "foo"'`

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java [hunk 4]
```diff
@@ -257,49 +255,53 @@ private PackageDescriptor getPackageDescriptor(TomlTableNode tomlTableNode) {
...
                     ProjectDiagnosticErrorCode.MISSING_PKG_INFO_IN_BALLERINA_TOML.diagnosticId(),
                     DiagnosticSeverity.WARNING, pkgNode.location());
-            org = defaultOrg().value();
         }
         name = getStringValueFromTomlTableNode(pkgNode, "name");
         if (name == null) {
-            reportDiagnostic("missing 'name' under [package]",
+            name = defaultName(this.projectPath).value();
+            reportDiagnostic("missing 'name' under [package], value derived as '" + name + "'",
```
Comment (sameerajayasoma) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java:295]: `missing key 'name' in table '[package]' in 'Ballerina.toml'. Defaulting to 'name = "bar"'`

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java [hunk 5]
```diff
@@ -257,49 +255,53 @@ private PackageDescriptor getPackageDescriptor(TomlTableNode tomlTableNode) {
...
                     ProjectDiagnosticErrorCode.MISSING_PKG_INFO_IN_BALLERINA_TOML.diagnosticId(),
                     DiagnosticSeverity.WARNING, pkgNode.location());
-            name = defaultName(this.projectPath).value();
         }
         version = getStringValueFromTomlTableNode(pkgNode, VERSION);
         if (version == null) {
-            reportDiagnostic("missing 'version' under [package]",
+            version = defaultVersion().value().toString();
+            reportDiagnostic("missing 'version' under [package], value derived as '" + version + "'",
```
Comment (sameerajayasoma) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java:302]: `missing key 'version' in table '[package]' in 'Ballerina.toml'. Defaulting to 'version = 0.1.0'`

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java [hunk 6]
```diff
@@ -257,49 +255,53 @@ private PackageDescriptor getPackageDescriptor(TomlTableNode tomlTableNode) {
         String name;
         String version;
 
+        String errorMessage = "missing package information in Ballerina.toml, values derived as org: '"
```
Comment (sameerajayasoma) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java:258]: Here is my suggestion:

```
Missing table `[package]` in `Ballerina.toml`. Defaulting to:
[package]
org = "foo"
name = "bar"
version = 1.0.0
```

**Review by sameerajayasoma — CHANGES_REQUESTED** [review]

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java (hunk 6 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java]: Refactored via https://github.com/ballerina-platform/ballerina-lang/pull/36174/commits/48fba38888323c8a03144c32f77884571964360f

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java (hunk 5 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java]: Refactored via https://github.com/ballerina-platform/ballerina-lang/pull/36174/commits/48fba38888323c8a03144c32f77884571964360f

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java (hunk 4 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java]: Refactored via https://github.com/ballerina-platform/ballerina-lang/pull/36174/commits/48fba38888323c8a03144c32f77884571964360f

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java (hunk 3 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java]: Refactored via https://github.com/ballerina-platform/ballerina-lang/pull/36174/commits/48fba38888323c8a03144c32f77884571964360f

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java (hunk 2 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java]: Removed in https://github.com/ballerina-platform/ballerina-lang/pull/36174/commits/48fba38888323c8a03144c32f77884571964360f

**Review by azinneera — APPROVED** [review]

**Pull request merged**